onnx/test/shape_inference_test.py

```diff
@@ -632,6 +633,15 @@ def test_unsqueeze_negative_axes(self) -> None:
             initializer=[make_tensor('axes', TensorProto.INT64, (2,), (0, -1))])
         self._assert_inferred(graph, [make_tensor_value_info('y', TensorProto.FLOAT, (1, 3, 4, 5, 1))])
 
+    def test_unsqueeze_scalar(self) -> None:
+        graph = self._make_graph(
+            [('x', TensorProto.FLOAT, ()),
+            ('axes', TensorProto.INT64, ())],
+            [make_node('Unsqueeze', ['x', 'axes'], 'y')],
+            [],
+            initializer=[make_tensor('axes', TensorProto.INT64, (), (-1,))])
+        self._assert_inferred(graph, [make_tensor_value_info('y', TensorProto.FLOAT, (1,))])
```

Comment:
Random thought: I found this test only works for UnSqueeze with opset_version >= 11 because older version does not support negative axes. To enhance test coverage, I suppose perhaps we can make every test in shape_inference_test.py be tested by every opset_version if the opset_version is not specified in the test.

Reply:
I think the issue (before and after version 11) is that axes was converted from attribute to input (not relating to negative values)? Anyway, on the other question: I don't see this test-case mention any opset_verion?

Reply:
> I think the issue (before and after version 11) is that axes was converted from attribute to input (not relating to negative values)?

It's because opset_version before 11 does not support negative axes.

> Anyway, on the other question: I don't see this test-case mention any opset_verion?

Without specified opset_version means using the latest opset_version. Going forward this test case should always be verified.

Regarding to the proposal here (make every test in shape_inference_test.py be tested by every opset_version if the opset_version is not specified in the test) May I understand your opinion about it @gramalingam ? Thanks.

Reply:
Hi Jacky, that's an interesting idea worth discussing (what the default behavior should be if no opset is specified), but I assume you mean that for a different PR, right? No need to mix that with this change for NonZero now, right?

Reply:
Yes, that would be a big change and should be done by another PR. We can discuss more details about it in our next meeting. I will keep this PR only focuses on fixing shape inference for NonZero.